libs-haskell/da-hs-base/src/Options/Applicative/Extended.hs

```diff
@@ -0,0 +1,23 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Options.Applicative.Extended
+    ( module Options.Applicative
+    , flagYesNoAuto
+    ) where
+
+import Options.Applicative
+
+
+-- | This constructs flags that can be set to yes, no, or auto to control a boolean value
+-- with auto using the default.
+flagYesNoAuto :: String -> Bool -> String -> Parser Bool
```

Comment:
I'm probably going to make a version of this that returns a type `Maybe Bool` at some point (`auto` being `Nothing`), so the default value can depend on other things like whether the current SDK version is less than the installed SDK version.